diff -Nur niffsdk1.02.orig/mk/bin.mk niffsdk1.02/mk/bin.mk
--- niffsdk1.02.orig/mk/bin.mk	2003-05-09 21:15:08.000000000 +0200
+++ niffsdk1.02/mk/bin.mk	2003-05-09 21:20:48.000000000 +0200
@@ -7,7 +7,7 @@
 $(BIN_TARGET): $(OBJS) $(DEPLIBS)
 	: $(RM) $(BIN_TARGET)
 	@echo Linking program $(BIN_TARGET)
-	$(CC) -o $(BIN_TARGET) $(OBJS) $(ALL_LDFLAGS) $(ALL_LIBS)
+	$(LIBTOOL) --mode=link $(CC) -o $(BIN_TARGET) $(OBJS) $(ALL_LDFLAGS) $(ALL_LIBS)
 
 clean::
 	$(RM) $(BIN_TARGET)
@@ -23,7 +23,7 @@
 install:: all
 	@if [ "$(EXPORT_PROGRAM)" ] ; then \
 		echo "Installing $(BIN_TARGET) in $(bindir)" ; \
-		$(INSTALL_PROGRAM) $(BIN_TARGET) $(DESTDIR)$(bindir) ; \
+		$(LIBTOOL) --mode=install $(INSTALL_PROGRAM) $(BIN_TARGET) $(DESTDIR)$(bindir) ; \
 	fi
 	
 uninstall:: 
diff -Nur niffsdk1.02.orig/mk/lib.mk niffsdk1.02/mk/lib.mk
--- niffsdk1.02.orig/mk/lib.mk	2003-05-09 21:15:08.000000000 +0200
+++ niffsdk1.02/mk/lib.mk	2003-05-09 21:25:19.000000000 +0200
@@ -1,15 +1,15 @@
 # Build a library
 
-LIB_FILENAME=$(EXPORT_LIBRARY)$(LIBRARY)$(LIB_SUFFIX)
+LIB_FILENAME=lib$(EXPORT_LIBRARY)$(LIBRARY)$(LIB_SUFFIX)
 LIB_TARGET=$(pkg_libdir)/$(LIB_FILENAME)
 
 all::$(LIB_TARGET)
 
-$(LIB_TARGET): $(OBJS)
+LOBJS = $(OBJS:o=lo)
+$(LIB_TARGET): $(LOBJS)
 	: $(RM) $(LIB_TARGET)
 	@echo Loading library $(LIB_TARGET)
-	$(AR) $(LIB_TARGET) $(OBJS)
-	$(RANLIB) $(LIB_TARGET)
+	$(LIBTOOL) --mode=link $(CC) -o $(LIB_TARGET) $(LOBJS) -rpath $(libdir) $(LIBRARY_LDFLAGS)
 
 clean::
 	$(RM) $(LIB_TARGET)
@@ -25,7 +25,7 @@
 install:: all
 	@if [ "$(EXPORT_LIBRARY)" ] ; then \
 		echo "Installing $(LIB_TARGET) in $(libdir)" ; \
-		$(INSTALL_DATA) $(LIB_TARGET) $(DESTDIR)$(libdir) ; \
+		$(LIBTOOL) --mode=install $(INSTALL_DATA) $(LIB_TARGET) $(DESTDIR)$(libdir) ; \
 	fi
 	
 uninstall:: 
diff -Nur niffsdk1.02.orig/mk/top.in niffsdk1.02/mk/top.in
--- niffsdk1.02.orig/mk/top.in	2003-05-09 21:15:08.000000000 +0200
+++ niffsdk1.02/mk/top.in	2003-05-09 21:27:33.000000000 +0200
@@ -8,7 +8,7 @@
 pkg_libdir=$(pkg_topdir)/lib
 pkg_bindir=$(pkg_topdir)/bin
 pkg_docdir=$(pkg_topdir)/doc
-
+top_builddir=$(pkg_topdir)
 
 all::
 
@@ -95,6 +95,7 @@
 PERL=@PERL@
 
 RANLIB=@RANLIB@
+LIBTOOL=@LIBTOOL@
 RM= rm -f
 YACC=@YACC@
 
@@ -111,7 +112,7 @@
 ALL_LIBS=$(DEPLIBS) $(LIBS) $(FLV_LIBS)
 
 
-LIB_SUFFIX=.a
+LIB_SUFFIX=.la
 BIN_SUFFIX=
 
 #OBJ_SUFFIX=.o
@@ -119,6 +120,9 @@
 .c.o:
 	$(CC) -c $(ALL_CPPFLAGS) $(ALL_CFLAGS) $<
 
+%.lo: %.c
+	$(LIBTOOL) --mode=compile $(CC) -c $(ALL_CPPFLAGS) $(ALL_CFLAGS) $<
+
 .SUFFIXES: .txt .pod .htm .man
 
 .c.txt:
diff -Nur niffsdk1.02.orig/niffio/configure.in niffsdk1.02/niffio/configure.in
--- niffsdk1.02.orig/niffio/configure.in	1996-06-12 03:57:45.000000000 +0200
+++ niffsdk1.02/niffio/configure.in	2003-05-09 21:29:10.000000000 +0200
@@ -5,7 +5,7 @@
 
 AC_PROG_CC
 AC_PROG_INSTALL
-AC_PROG_RANLIB
+AC_PROG_LIBTOOL
 
 AC_PATH_PROGS(MAKEDEPEND, makedepend, :)
 
diff -Nur niffsdk1.02.orig/niffio/local.in niffsdk1.02/niffio/local.in
--- niffsdk1.02.orig/niffio/local.in	2003-05-09 21:15:09.000000000 +0200
+++ niffsdk1.02/niffio/local.in	2003-05-09 21:29:54.000000000 +0200
@@ -6,4 +6,4 @@
 riffio_incdir=${pkg_topdir}/../riffio/include
 
 PKG_CPPFLAGS=-I$(pkg_incdir) -I$(niff_incdir) -I$(riffio_incdir)
-PKG_DEPLIBS=$(pkg_libdir)/niffio.a $(riffio_libdir)/stdcriff.a $(riffio_libdir)/riffio.a  
+PKG_DEPLIBS=$(pkg_libdir)/libniffio.la $(riffio_libdir)/libstdcriff.la $(riffio_libdir)/libriffio.la  
diff -Nur niffsdk1.02.orig/niffio/src/lib/niffio/Makefile niffsdk1.02/niffio/src/lib/niffio/Makefile
--- niffsdk1.02.orig/niffio/src/lib/niffio/Makefile	1996-06-13 07:11:08.000000000 +0200
+++ niffsdk1.02/niffio/src/lib/niffio/Makefile	2003-05-09 21:34:25.000000000 +0200
@@ -6,6 +6,7 @@
 
 EXPORT_LIBRARY = niffio
 EXPORT_HEADERS = $(pkg_incdir)/niffio.h
+LIBRARY_LDFLAGS = ../../../../riffio/lib/libriffio.la ../../../../riffio/lib/libstdcriff.la
 
 SRCS = \
 clt.c\
diff -Nur niffsdk1.02.orig/riffio/configure.in niffsdk1.02/riffio/configure.in
--- niffsdk1.02.orig/riffio/configure.in	1996-06-06 08:09:37.000000000 +0200
+++ niffsdk1.02/riffio/configure.in	2003-05-09 21:29:19.000000000 +0200
@@ -5,7 +5,7 @@
 
 AC_PROG_CC
 AC_PROG_INSTALL
-AC_PROG_RANLIB
+AC_PROG_LIBTOOL
 
 AC_PATH_PROGS(MAKEDEPEND, makedepend, :)
 
diff -Nur niffsdk1.02.orig/riffio/local.mk niffsdk1.02/riffio/local.mk
--- niffsdk1.02.orig/riffio/local.mk	1996-06-06 04:06:50.000000000 +0200
+++ niffsdk1.02/riffio/local.mk	2003-05-09 21:30:06.000000000 +0200
@@ -3,4 +3,4 @@
 # Package specific make 
 
 PKG_CPPFLAGS=-I$(pkg_incdir)
-PKG_DEPLIBS=$(pkg_libdir)/stdcriff.a $(pkg_libdir)/riffio.a  
+PKG_DEPLIBS=$(pkg_libdir)/libstdcriff.la $(pkg_libdir)/libriffio.la  
